packages/x-data-grid-premium/src/components/sidebar/columnPanel/GridSidebarColumnPanelPivotMenu.tsx

```diff
@@ -0,0 +1,187 @@
+  const rootProps = useGridRootProps();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const isAvailableField = modelKey === null;
+  const fieldIndexInModel = !isAvailableField
+    ? props.pivotModel[modelKey].findIndex((item) => item.field === field)
+    : -1;
+  const modelLength = !isAvailableField ? props.pivotModel[modelKey].length : 0;
+  const canMoveUp = fieldIndexInModel > 0;
+  const canMoveDown = !isAvailableField && fieldIndexInModel < modelLength - 1;
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleMove = (to: 'up' | 'down' | 'top' | 'bottom' | FieldTransferObject['modelKey']) => {
+    handleClose();
+
+    // Do nothing if the field is already in the target section
+    if (to === modelKey) {
+      return;
+    }
+
+    let targetField: string | undefined;
+    let targetFieldPosition: DropPosition = null;
+    let targetSection: FieldTransferObject['modelKey'] = modelKey;
+
+    switch (to) {
+      case 'up':
+        targetField = props.pivotModel[modelKey!][fieldIndexInModel - 1].field;
+        targetFieldPosition = 'top';
+        break;
+      case 'down':
+        targetField = props.pivotModel[modelKey!][fieldIndexInModel + 1].field;
+        targetFieldPosition = 'bottom';
+        break;
+      case 'top':
+        targetField = props.pivotModel[modelKey!][0].field;
+        targetFieldPosition = 'top';
+        break;
+      case 'bottom':
+        targetField = props.pivotModel[modelKey!][modelLength - 1].field;
+        targetFieldPosition = 'bottom';
+        break;
+      case 'rows':
+      case 'columns':
+      case 'values':
+      case null:
+        targetSection = to;
+        break;
+      default:
+        break;
+    }
+
+    updatePivotModel({
+      field,
+      targetField,
+      targetFieldPosition,
+      targetSection,
+      originSection: modelKey,
+    });
+  };
+
+  const menuProps = {
+    id: 'pivot-menu',
+    anchorEl,
+    open,
+    onClose: handleClose,
+    MenuListProps: {
+      'aria-labelledby': 'pivot-menu-button',
+    },
+  };
+
+  return (
+    <div>
+      <rootProps.slots.baseIconButton
+        size="small"
+        {...rootProps.slotProps?.baseIconButton}
+        id="pivot-menu-button"
+        aria-controls={open ? 'pivot-menu' : undefined}
+        aria-haspopup="true"
+        aria-expanded={open ? 'true' : undefined}
+        onClick={handleClick}
+      >
+        {isAvailableField ? (
+          <rootProps.slots.pivotMenuAddIcon fontSize="small" />
+        ) : (
+          <rootProps.slots.columnMenuIcon fontSize="small" />
+        )}
+      </rootProps.slots.baseIconButton>
+
+      {isAvailableField ? (
+        <Menu {...menuProps}>
+          <MenuItem dense onClick={() => handleMove('rows')}>
+            Add to Rows
```

Comment:
Seems to be a bug with this action:

https://github.com/user-attachments/assets/7272d53c-783c-47eb-b6ef-8ac09ac1d916

Reply:
Yes, that's a regression I introduced recently.
I'll fix it 👌🏼